### ISSUE TITLE
Button or menu option in editor for adding a dictionary key/value with types matching previous entry

### ISSUE THREAD

**Issue by Rubrickus** [godotengine/godot-proposals#5887]
### Describe the project you are working on

Very early stages of an eventual RPG prototype. (Mostly just learning Godot 4.)

### Describe the problem or limitation you are having in your project

I have various game objects, and I want each to have an open-ended set of "traits", e.g. "squishiness: 0.7". A dictionary is the obvious choice for this. However, adding multiple new key/value pairs to an exported Dictionary in the editor is quite tedious and painful, since you have to select the type of each key and value individually.

https://github.com/godotengine/godot-proposals/issues/56 looks to solve this with typed dictionaries, but that's a tricky proposal which isn't currently being worked on. This proposal is for an editor-level partial workaround.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

A modification/addition to the "Add Key/Value Pair" button (or maybe a contextual menu?) for "Add Key/Value Pair With Same Types as Previous Entry. (I'll admit I don't know how to convey that succinctly in text!). The user would then only have to add one entry the "hard" way, and then essentially duplicate the type info from the previous entry.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Not sure the best UI, but a kebab menu or contextual menu on the "Add Key/Value Pair" button seems most likely.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Not really. (Likeliest workaround is to edit the raw scene files directly.)

### Is there a reason why this should be core and not an add-on in the asset library?

It's an editor feature. (I haven't tried writing any editor plugins yet, dunno if this sort of thing would be doable in that form).

**Comment by Zireael07** [comment]
Did you see #5775? Unfortunately current core Godot is a bit clunky when it comes to editing Dictionaries in inspector

**Comment by Mickeon** [comment]
I agree on the issue, but not necessarily on the solution. I suppose https://github.com/godotengine/godot-proposals/issues/5775 would improve the feature already in a more tidily organised manner.

**Comment by Rubrickus** [comment]
I hadn't seen #5775. That would indeed cover this request and more!
